src/session_impl.cpp

```diff
@@ -697,15 +704,15 @@ namespace aux {
 			if (val) m_settings.set_int(settings_pack::allowed_enc_level, val.int_value());
 		}
 #endif
-
+		bool need_apply = false;
 		if (flags & session::save_settings)
 		{
-			settings = e->dict_find_dict("settings");
+			settings = e.dict_find_dict("settings");
 			if (settings)
 			{
+				load_pack_from_dict(settings, pack);
 				// apply_settings_pack will update dht and proxy
-				std::shared_ptr<settings_pack> pack = load_pack_from_dict(settings);
-				apply_settings_pack(pack);
+				need_apply = true;
```

Comment:
hm.. but you only apply it if we load it from the settings, so anything passed in through it would be ignored, unless loaded from the settings file..